File: androidApp/src/main/java/dev/emanuelepapa/kmmsurvivalguidesamples/android/result/ResultScreen.kt
package dev.emanuelepapa.kmmsurvivalguidesamples.android.result

import androidx.compose.foundation.layout.Column
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState

@Composable
fun ResultScreen(viewModel: ResultScreenViewModel) {
    val state = viewModel.state.collectAsState().value
    Column {
        Button(onClick = { viewModel.onShowRandomDoubleClicked() }) {
            Text(text = "Show random double")
        }
        Text(text = state.formattedRandomDouble ?: "Not generated yet")
    }
}

File: iosApp/iosApp/result/ResultScreen.swift
//
//  ResultScreen.swift
//  iosApp
//
//  Created by Emanuele on 17/09/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import shared

struct ResultScreen: View {
    
    @ObservedObject var viewModel: ResultScreenViewModel
    
    var body: some View {
        VStack {
            Button {
                viewModel.onShowRandomDoubleClicked()
            } label: {
                Text("Show random double")
            }
            Text(viewModel.state.formattedRandomDouble ?? "Not generated yet")
        }
    }
}
